packages/protvista-datatable/src/protvista-datatable.ts

```diff
@@ -262,9 +300,42 @@ class ProtvistaDatatable extends LitElement {
     );
   }
 
+  handleFilterChange(e: Event, filterName: string): void {
+    const { selectedOptions } = e.target as HTMLSelectElement;
+    // Only 1 can be selected
+    const { value } = selectedOptions.item(0);
+    if (value) {
+      this.selectedFilters.set(filterName, value);
+    } else {
+      this.selectedFilters.delete(filterName);
+    }
+    this.updateRowStyling();
+  }
+
+  isRowVisible(row: HTMLTableRowElement): boolean {
+    if (!this.selectedFilters || this.selectedFilters.size <= 0) {
+      return true;
+    }
+    let hasMatch = false;
+    this.selectedFilters.forEach((value, filterName) => {
+      const column = row.querySelector(`[data-filter="${filterName}"]`);
+      if (column && column.innerHTML === value) {
+        hasMatch = true;
+      }
+    });
+    return hasMatch;
+  }
+
   updateRowStyling(): void {
     let oddOrEvenCount = 0;
     this.rows?.forEach((row) => {
+      // Filter visibility
+      const isRowVisible = this.isRowVisible(row);
```

Comment:
Could be a little easier to understand if all of the determination (including L385-395) of a row being visible is within `isRowVisible`

Reply:
Moved some of the logic out.